.github/workflows/differential-shellcheck.yaml

```diff
@@ -0,0 +1,32 @@
+---
+# https://github.com/redhat-plumbers-in-action/differential-shellcheck#readme
+
+name: Differential ShellCheck
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  contents: read
```

Comment:
I think you can change this to `write` and drop lines 20-21?

Reply:
You raise an interesting point about lines 20-21. However, these permissions serve different purposes according to the GitHub documentation:

The recommended configuration distinguishes between two key permission scopes:
- `contents: read` allows viewing repository contents
- `actions: write` enables modifying workflow runs and their status

These aren't redundant but have distinct functions. As detailed in the [GitHub docs](https://docs.github.com/en/enterprise-server@3.11/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_-token), each permission grants specific capabilities.

quote from the linked document:

| Permission | Allows an action using GITHUB_TOKEN to |
| --- | --- |
| `content` |  Work with the contents of the repository. For example, contents: read permits an action to list the commits, and contents: write allows the action to create a release. For more information, see "[Permissions required for GitHub Apps](https://docs.github.com/en/enterprise-server@3.11/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents)."
| `security-events` | Work with GitHub code scanning and Dependabot alerts. For example, security-events: read permits an action to list the Dependabot alerts for the repository, and security-events: write allows an action to update the status of a code scanning alert. For more information, see "[Repository permissions for 'Code scanning alerts'](https://docs.github.com/en/enterprise-server@3.11/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-code-scanning-alerts)" and "[Repository permissions for 'Dependabot alerts'](https://docs.github.com/en/enterprise-server@3.11/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-dependabot-alerts)" in "Permissions required for GitHub Apps." |

also, I'm implementing the recommended configuration from the Differential ShellCheck GitHub Action, with some modifications tailored to our specific project requirements based on https://github.com/redhat-plumbers-in-action/differential-shellcheck?tab=readme-ov-file#usage .